pages/docs/middlewares.en-US.md

```diff
@@ -15,72 +15,99 @@ The `middlewares` option is a new addition in SWR 1.0 that enables you to execut
 
 ```jsx
-const middleware = (useSWRNext) => (key, fetcher, config) => {
-  // Before hook runs...
-  const swr = useSWRNext(key, fetcher, config)
-  // After hook runs...
-  return swr
+function myMiddleware (useSWRNext) {
+  return (key, fetcher, config) => {
+    // Before hook runs...
+    const swr = useSWRNext(key, fetcher, config)
+    // After hook runs...
+    return swr
+  }
 }
 ```
 
-Pass middleware as an array to any `useSWR` hook.
+You can pass an array of middleware as an option to `SWRConfig` or `useSWR`:
 
-## Example
+```jsx
+<SWRConfig value={{ middleware: [myMiddleware] }}>
 
-### Request Logger
+// or...
 
-Let's build a simple request logger middleware as an example. It simply prints out all the fetcher requests sent from the SWR hook.
+useSWR(key, fetcher, { middleware: [myMiddleware] })
+```
 
+### Extend
 
-```jsx
-import useSWR from 'swr'
+Middleware will be extended like regular options. For example:
 
-const fetcher (key) => fetch(`/api?key=${key}`).then(res => res.text())
-
-const loggerMiddleware = (useSWRNext) => (key, fetcher, config) => {
-  console.log(`Request to /api with param key=${key}`)
-  return useSWRNext(key, fetcher, config)
+```jsx
+function Bar () {
+  useSWR(key, fetcher, { middleware: [c] })
+  // ...
 }
 
-useSWR(key, fetcher, {
-  middlewares: [loggerMiddleware]
-})
+function Foo() {
+  return (
+    <SWRConfig value={{ middleware: [a] }}>
+      <SWRConfig value={{ middleware: [b] }}>
+        <Bar/>
+      </SWRConfig>
+    </SWRConfig>
+  )
+}
 ```
 
-Every time the SWR hook runs, it will output to the console:
+is equivalent to:
 
-```
-Request to /api with param key=key1
-Request to /api with param key=key2
-Request to /api with param key=key3
+```js
+useSWR(key, fetcher, { middleware: [a, b, c] })
 ```
 
-### Nested Middlewares
+### Multiple Middleware
 
-Middlwares are applied on top of each other, as commonly found in other libraries. If you pass multiple middleware to one hook:
+Each middleware wraps the next middleware, and the last one just wraps the SWR hook. For example:
 
 ```jsx
-useSWR(key, fetcher, {
-  middlewares: [A, B, C]
-})
+useSWR(key, fetcher, { middleware: [a, b, c] })
 ```
 
-The order of execution will be `A → B → C`, as shown below:
+The order of middleware executions will be `a → b → c`, as shown below:
+
 ```
 enter A
   enter B
     enter C
+      useSWR()
     exit  C
   exit  B
 exit  A
 ```
 
-If you have any middleware configured in `SWRConfig`, the outermost middleware will run first. In the following example, the order will be `A → B`:
+## Examples
+
+### Request Logger
+
+Let's build a simple request logger middleware as an example. It simply prints out all the fetcher requests sent from the SWR hook.
+
+
+```jsx
+import useSWR from 'swr'
+
+const fetcher (key) => fetch(`/api?key=${key}`).then(res => res.text())
+
+const loggerMiddleware = (useSWRNext) => (key, fetcher, config) => {
```

Comment:
Better to keep consistence to either use named functions or arrow functions. The former one if you feel middleware function name might be import (like for devtools)

Reply:
I think it's easier for people who are not an JS expert to understand this:

```js
function A() {
  return () => {}
}
```

Than:

```js
const A = () => () => {}
```